src/storage/storage_structure/in_mem_file.cpp

```diff
@@ -35,32 +35,56 @@ ku_string_t InMemFile::appendString(const char* rawString) {
     std::memcpy(result.prefix, rawString,
         length <= ku_string_t::SHORT_STR_LENGTH ? length : ku_string_t::PREFIX_LENGTH);
     if (length > ku_string_t::SHORT_STR_LENGTH) {
-        if (length > BufferPoolConstants::PAGE_4KB_SIZE) {
+        if (length > BufferPoolConstants::PAGE_256KB_SIZE) {
             throw CopyException(ExceptionMessage::overLargeStringPKValueException(length));
         }
-        // Allocate a new page if necessary.
-        if (nextOffsetInPageToAppend + length >= BufferPoolConstants::PAGE_4KB_SIZE) {
-            addANewPage();
-            nextOffsetInPageToAppend = 0;
-            nextPageIdxToAppend++;
+
+        int32_t remainingLength = length;
+        while (remainingLength > 0) {
+            // Allocate a new page if necessary.
+            if (nextOffsetInPageToAppend + length >= BufferPoolConstants::PAGE_4KB_SIZE) {
+                addANewPage();
+                nextOffsetInPageToAppend = 0;
+                nextPageIdxToAppend++;
+            }
+            auto numBytesToWriteInPage = std::min(static_cast<uint64_t>(remainingLength),
+                BufferPoolConstants::PAGE_4KB_SIZE - nextOffsetInPageToAppend);
+            pages[nextPageIdxToAppend]->write(nextOffsetInPageToAppend,
+                (uint8_t*)rawString + (length - remainingLength), numBytesToWriteInPage);
+            // Only do this the first time
+            if (static_cast<uint64_t>(remainingLength) == length) {
+                TypeUtils::encodeOverflowPtr(
```

Comment:
You seem to be leaving gaps here, which is inconsistent from `DiskOverflowFile`. I don't think it's necessary to leave gaps anymore.

Reply:
I've fixed that bug, and it revealed that I was missing changes to the HashIndexBuilder's in-memory lookup.